src/Security/MemberAuthenticator/MemberAuthenticator.php

```diff
@@ -94,7 +94,13 @@ protected function authenticateMember($data, ValidationResult &$result = null, M
         } elseif (!$asDefaultAdmin) {
             // spoof a login attempt
             $tempMember = Member::create();
-            $tempMember->{Member::config()->get('unique_identifier_field')} = $email;
+
+            if($uniqueIdentifierFieldName != 'Email')
+                $tempMember->$uniqueIdentifierFieldName = $username;
+
+            if(!empty(DefaultAdminService::getDefaultAdminEmail()))
+                $tempMember->Email = DefaultAdminService::getDefaultAdminEmail();
```

Comment:
Why are we using the default admin email here? This logic path is explicitly for when we are *not* authenticating as default admin